### PR TITLE
Amp-bind Clarify bind expression errors

### LINKED ISSUE #8251: includes is not available for amp-bind
amp-bind should support `includes` as part  of [7401](https://github.com/ampproject/amphtml/pull/7401), but I get:

> error.js:135 AMP.setState() failed with error: : includes is not a supported function.​​​
Dd @ error.js:135
f.error @ log.js:214
(anonymous) @ bind-impl.js:183

You can reproduce the issue by visiting:
[product sample](https://amp-by-example-staging.appspot.com/samples_templates/product/preview/)

@kmh287

### PR DESCRIPTION
**Pull request by kmh287**
If we don't have any functions whitelisted for a particular type, the type information isn't included in an error message. Since we don't whitelist any methods on Object, any attempt to call a method on an object provides an error message without statign the caller type. This caused the confusion behind #8251 

This PR ensures that any call to a non-built in function will include the caller type (even if it's null) to make debugging easier. For example, the issue from #8251 will show the error:

`[object Object].includes is not a supported function.`

instead of

`  includes is not a supported function.`

Fixes #8251 

/to @choumx

### CODE REVIEW

File: extensions/amp-bind/0.1/bind-expression.js [hunk 1]
```diff
@@ -139,23 +139,24 @@ export class BindExpression {
         const isBuiltIn = (args[0] === null);
 
         const caller = this.eval_(args[0], scope);
+        const callerType = Object.prototype.toString.call(caller);
```
Comment (choumx) [inline extensions/amp-bind/0.1/bind-expression.js]: This is only used in the `else` block below. We should keep vars within the minimal scope and avoid doing this work unnecessarily when the invocation is on a built-in function.

File: extensions/amp-bind/0.1/bind-expression.js [hunk 2]
```diff
@@ -139,23 +139,24 @@ export class BindExpression {
...
+        let validFunction;
+        let unsupportedError;
 
         if (isBuiltIn) {
+          unsupportedError = `${method} is not a supported function.`;
           validFunction = FUNCTION_WHITELIST[BUILT_IN_FUNCTIONS][method];
         } else {
-          const callerType = Object.prototype.toString.call(caller);
+          unsupportedError =
+              `${callerType}.${method} is not a supported function.`;
```
Comment (choumx) [inline extensions/amp-bind/0.1/bind-expression.js]: Nit: Move below the `if (whitelist)` conditional and check for non-existence of `validFunction`.

File: extensions/amp-bind/0.1/bind-expression.js (hunk 1 above)
Comment (kmh287) [inline extensions/amp-bind/0.1/bind-expression.js]: done

File: extensions/amp-bind/0.1/bind-expression.js (hunk 2 above)
Comment (kmh287) [inline extensions/amp-bind/0.1/bind-expression.js]: done

File: extensions/amp-bind/0.1/bind-expression.js [hunk 3]
```diff
@@ -142,10 +142,11 @@ export class BindExpression {
         const params = this.eval_(args[1], scope);
         const method = String(value);
 
-        let validFunction = null;
-        let unsupportedError = `${method} is not a supported function.`;
+        let validFunction;
+        let unsupportedError;
 
         if (isBuiltIn) {
+          unsupportedError = `${method} is not a supported function.`;
```
Comment (choumx) [inline extensions/amp-bind/0.1/bind-expression.js]: Nit: Do the same for this case.

Comment (kmh287) [inline extensions/amp-bind/0.1/bind-expression.js]: done, but I don't feel this is as readable as before.

**Pull request merged**